tests/test_minimal/test_import_structure.py

```diff
@@ -32,10 +33,25 @@ def test_outer_import_structure(self):
             "utils",
             # Exposed attributes
             "NWBConverter",
+            "run_conversion_from_yaml",
+        ]
+        self.assertCountEqual(first=current_structure, second=expected_structure)
+
+    def test_tools_import_structure(self):
+        current_structure = _strip_magic_module_attributes(dictionary=tools.__dict__)
+        expected_structure = [
+            # Sub-modules
+            "neo",
             "spikeinterface",
             "roiextractors",
-            "neo",
-            "run_conversion_from_yaml",
+            # Helper functions
+            "get_package",
+            "importing",
+            # Other stuff that I don't know why it is here (for PR)
+            "nwb_helpers",
```

Comment:
What about this stuff here? Why are they imported with `from neuroconv import tools`?

Reply:
Half of these are package structure available from all import navigation (those are sub-modules). Can't hide those without `__dir__` override magic.

But this test is good, definitely keep it and keep it up to date with what we intend the package structure to appear as.

Reply:
My question was more like, when are they imported? the init of tools does not have them and neither does the top level init.

Reply:
If you call `import neuroconv`, only the names of the submodules are loaded into the `__dir__`, not their contents. So actually trigger import code in the submodule you'd have to call `neuroconv.tools.submodule.some_function` and it will load that entire submodule. That's the classic way of achieving 'lazy' imports that Bens original solution utilized (albeit at deeper levels of nesting).

Reply:
Alternative way to trigger imports there is `from neuroconv.tools import submodule`

Reply:
Ok so my understanding is the following.  When I call `from neuroconv import tools`, "nwb_helpers" is added to tools.__dict__  because of two conditions:
1) nwb_helpers is a submodule of tools 
AND
2) Some functions form nwb_helpers are called in the import hierarchy. What I mean is that the import `NWBConverter` in the top __init__ here triggers this lines here:
https://github.com/catalystneuro/neuroconv/blob/ae08b4ae627f9e5ced0e0d59d15c1bd329abffb8/src/neuroconv/nwbconverter.py#L10

Reply:
Yes, that is true as well